utils/common/src/main/java/org/apache/brooklyn/util/text/VersionComparator.java

```diff
@@ -56,149 +56,69 @@ public static boolean isSnapshot(String version) {
-            String v1p = v1Parts[i];
-            String v2p = v2Parts[i];
-            
-            if (v1p.equals(v2p)) continue;
-            
-            if (isNumberInFirstChar(v1p) || isNumberInFirstChar(v2p)) {
-                // something starting with a number is higher than something not
-                if (!isNumberInFirstChar(v1p)) return -1;
-                if (!isNumberInFirstChar(v2p)) return 1;
-                
-                // both start with numbers; can use natural order comparison *unless*
-                // one is purely a number AND the other *begins* with that number,
-                // followed by non-digit chars, in which case prefer the pure number
-                // ie:
-                //           1beta < 1
-                // but note
-                //            1 < 2beta < 11beta
-                if (isNumber(v1p) || isNumber(v2p)) {
-                    if (!isNumber(v1p)) {
-                        if (v1p.startsWith(v2p)) {
-                            if (!isNumberInFirstChar(Strings.removeFromStart(v1p, v2p))) {
-                                // v2 is a number, and v1 is the same followed by non-numbers
-                                return -1;
-                            }
-                        }
-                    }
-                    if (!isNumber(v2p)) {
-                        // as above but inverted
-                        if (v2p.startsWith(v1p)) {
-                            if (!isNumberInFirstChar(Strings.removeFromStart(v2p, v1p))) {
-                                return 1;
-                            }
-                        }
-                    }
-                    // both numbers, skip to natural order comparison
-                }
-            }
-            
-            // otherwise it is in-order
-            int result = NaturalOrderComparator.INSTANCE.compare(v1p, v2p);
-            if (result!=0) return result;
-        }
-    }
+        // qualified items are lower than unqualified items
+        TwoBooleans no_qualifier = TwoBooleans.of(u1.equals(v1), u2.equals(v2));
+        if (no_qualifier.different()) return no_qualifier.compare(false);
 
-    @VisibleForTesting
-    static String[] splitOnNonWordChar(String v) {
-        Collection<String> parts = new ArrayList<String>();
-        String remaining = v;
+        // now compare qualifiers
+        
+        // allow -, ., and _ as qualifier offsets; if used, compare qualifer without that
+        String q1 = v1.substring(u1.length());
+        String q2 = v2.substring(u2.length());
+        
+        String qq1 = q1, qq2 = q2;
+        if (q1.startsWith("-") || q1.startsWith(".") || q1.startsWith("_")) q1 = q1.substring(1);
+        if (q2.startsWith("-") || q2.startsWith(".") || q2.startsWith("_")) q2 = q2.substring(1);
+        uq = NaturalOrderComparator.INSTANCE.compare(q1, q2);
+        if (uq!=0) return uq;
+        
+        // if qualifiers the same, look at qualifier separator char
```

Comment:
as above if someone requests a comparison between different strings we should give one.  it's a perverse case but equality determination should be done by someone else.